src/components/biggive-carousel/biggive-carousel.tsx

```diff
@@ -0,0 +1,103 @@
+import { Component, Prop, Element, h } from '@stencil/core';
+import { brandColour } from '../../globals/brand-colour';
+import { spacingOption } from '../../globals/spacing-option';
+
+@Component({
+  tag: 'biggive-carousel',
+  styleUrl: 'biggive-carousel.scss',
+  shadow: true,
+})
+export class BiggiveCarousel {
+  @Element() host: HTMLBiggiveCarouselElement;
+
+  @Prop() spaceBelow: spacingOption = 4;
+
+  @Prop() columnCount: number = 3;
+
+  @Prop() buttonBackgroundColour: brandColour = 'white';
+
+  @Prop() buttonIconColour: brandColour = 'primary';
+
+  currentTab = 0;
+  itemCount = 0;
+  itemWidth = 0;
+  max = 0;
+  min = 0;
+  sleeve;
+
+  componentDidRender() {
+    this.sleeve = this.host.shadowRoot?.querySelector<HTMLElement>('.sleeve');
+    let children: Array<any> = Array.from(this.host.children);
+
+    this.itemCount = children.length;
+
+    if (children.length > 0 && this.sleeve) {
```

Comment:
Are we actually expecting we might not have a sleave, or is the `&& this.sleeve` part just here to satisfy the compiler?

If we should always have a sleave then instead of putting it in a conditional like this would it be better to either use a type assertion to check let the compiler assume its there, or a guard clause to explicitly check it and throw if its missing?

Reply:
There will always be a a sleeve. I have updated the code to remove this check